src/libYARP_OS/src/PortCoreInputUnit.cpp

```diff
@@ -156,103 +156,103 @@ void PortCoreInputUnit::run() {
         InputStream *is = &ip->getInputStream();
         is->setReadEnvelopeCallback(envelopeReadCallback, this);
     }
+    if (ip != YARP_NULLPTR) {
+        while (!done) {
+            ConnectionReader& br = ip->beginRead();
+
+            if (br.getReference() != YARP_NULLPTR) {
+                //printf("HAVE A REFERENCE\n");
+                if (localReader != YARP_NULLPTR) {
+                    bool ok = localReader->read(br);
+                    if (!br.isActive()) { break; }
+                    if (!ok) continue;
+                }
+                else {
+                    PortManager& man = getOwner();
+                    bool ok = man.readBlock(br, id, YARP_NULLPTR);
+                    if (!br.isActive()) { break; }
+                    if (!ok) continue;
+                }
+                //printf("DONE WITH A REFERENCE\n");
+                ip->endRead();
+                continue;
+            }
```

Comment:
Please save this big "coverity fix - dereference before null check - PortCoreInputUnit.cpp" commit for later